[2025-09-20T18:58:48.455+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-09-20T18:58:48.492+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_fact_Forecast scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-09-20T18:58:48.514+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_fact_Forecast scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-09-20T18:58:48.515+0000] {taskinstance.py:2867} INFO - Starting attempt 3 of 3
[2025-09-20T18:58:48.545+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_load_fact_Forecast> on 2024-01-01 00:00:00+00:00
[2025-09-20T18:58:48.554+0000] {standard_task_runner.py:72} INFO - Started process 163 to run task
[2025-09-20T18:58:48.561+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'bronze_full_load_spaceparts_optimized', 'extract_load_fact_Forecast', 'scheduled__2024-01-01T00:00:00+00:00', '--job-id', '167', '--raw', '--subdir', 'DAGS_FOLDER/01_Bronce.py', '--cfg-path', '/tmp/tmpyku7rjc8']
[2025-09-20T18:58:48.567+0000] {standard_task_runner.py:105} INFO - Job 167: Subtask extract_load_fact_Forecast
[2025-09-20T18:58:48.942+0000] {task_command.py:467} INFO - Running <TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_fact_Forecast scheduled__2024-01-01T00:00:00+00:00 [running]> on host 050d5b6b4fc2
[2025-09-20T18:58:49.052+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-team' AIRFLOW_CTX_DAG_ID='bronze_full_load_spaceparts_optimized' AIRFLOW_CTX_TASK_ID='extract_load_fact_Forecast' AIRFLOW_CTX_EXECUTION_DATE='2024-01-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-01T00:00:00+00:00'
[2025-09-20T18:58:49.054+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-09-20T18:58:49.070+0000] {01_Bronce.py:385} INFO - Starting optimized extraction for table: fact.Forecast
[2025-09-20T18:58:52.202+0000] {01_Bronce.py:398} INFO - Table fact.Forecast has 5,197 rows
[2025-09-20T18:58:52.203+0000] {01_Bronce.py:420} INFO - Using chunk size: 50,000 for 5,197 rows
[2025-09-20T18:58:52.204+0000] {01_Bronce.py:425} INFO - Starting streaming extraction and load for fact.Forecast
[2025-09-20T18:58:55.643+0000] {01_Bronce.py:235} INFO - Processing chunk 1 with 5197 rows
[2025-09-20T18:58:55.695+0000] {01_Bronce.py:298} ERROR - Error creating table bronze_fact_forecast: 'Connection' object has no attribute 'commit'
[2025-09-20T18:58:55.695+0000] {01_Bronce.py:360} ERROR - Error in streaming load to bronze_fact_forecast: 'Connection' object has no attribute 'commit'
[2025-09-20T18:58:55.696+0000] {01_Bronce.py:445} ERROR - Error processing table fact.Forecast: 'Connection' object has no attribute 'commit'
[2025-09-20T18:58:55.728+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/01_Bronce.py", line 427, in extract_and_load_table_optimized
    records_loaded = loader.load_data_streaming(data_iterator, bronze_table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 343, in load_data_streaming
    column_mapping = self.create_table_if_not_exists(chunk, table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 292, in create_table_if_not_exists
    conn.commit()
AttributeError: 'Connection' object has no attribute 'commit'
[2025-09-20T18:58:55.737+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=bronze_full_load_spaceparts_optimized, task_id=extract_load_fact_Forecast, run_id=scheduled__2024-01-01T00:00:00+00:00, execution_date=20240101T000000, start_date=20250920T185848, end_date=20250920T185855
[2025-09-20T18:58:55.756+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-09-20T18:58:55.756+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 167 for task extract_load_fact_Forecast ('Connection' object has no attribute 'commit'; 163)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/01_Bronce.py", line 427, in extract_and_load_table_optimized
    records_loaded = loader.load_data_streaming(data_iterator, bronze_table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 343, in load_data_streaming
    column_mapping = self.create_table_if_not_exists(chunk, table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 292, in create_table_if_not_exists
    conn.commit()
AttributeError: 'Connection' object has no attribute 'commit'
[2025-09-20T18:58:55.790+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-09-20T18:58:55.889+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-09-20T18:58:55.893+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
